RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -459,6 +473,7 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
     // link calo clusters
     linkCaloClusters(ootSuperCluster, *ootEbeeClusters, ootEbeeClusterMap, rechitMap, barrelHitHandle, endcapHitHandle, caloTopology, *ootEsClusters, ootEsClusterMap);
   }
+  index = 0;
```

Comment:
as above: reset it only right before the loop where it's needed instead of right after the last loop where it was used